File: WhatIsGoodWebApp/Domain/Repositories/IEventRepository.cs
using System;
using System.Collections.Generic;
using WhatsGood.Domain.Entities;


namespace WhatsGood.Domain.Repositories
{
    public interface IEventRepository : IBaseRepository<Event>
    {
        IEnumerable<Event> GetByStartDate(DateTime startDate);
        IEnumerable<Event> GetByGenre(Genre genre);
        IEnumerable<Event> GetNextEvents();
        IEnumerable<Event> GetByGenre(string genreDescription);
    }
}

File: WhatsGoodAndroidApp/src/com/dscode/whatsgood/model/IEventRepository.java
package com.dscode.whatsgood.model;

import java.util.Date;
import java.util.List;

public interface IEventRepository {	
	List<Event> GetByDate(Date startDate);
	List<Event> GetByGenre(String genreDescription);
	List<Event> GetByArtist(String artistName);
}
